include/nlohmann/json.hpp

```diff
@@ -5644,28 +5745,37 @@ class basic_json
         // insert only works for arrays
         if (JSON_HEDLEY_UNLIKELY(!is_array()))
         {
-            JSON_THROW(type_error::create(309, "cannot use insert() with " + std::string(type_name())));
+            JSON_THROW(type_error::create(309, "cannot use insert() with " + std::string(type_name()), diagnostics_t(*this)));
         }
 
         // check if iterator pos fits to this JSON value
         if (JSON_HEDLEY_UNLIKELY(pos.m_object != this))
         {
-            JSON_THROW(invalid_iterator::create(202, "iterator does not fit current value"));
+            JSON_THROW(invalid_iterator::create(202, "iterator does not fit current value", diagnostics_t(*this)));
         }
 
         // check if range iterators belong to the same JSON object
         if (JSON_HEDLEY_UNLIKELY(first.m_object != last.m_object))
         {
-            JSON_THROW(invalid_iterator::create(210, "iterators do not fit"));
+            JSON_THROW(invalid_iterator::create(210, "iterators do not fit", diagnostics_t(*this)));
         }
 
         if (JSON_HEDLEY_UNLIKELY(first.m_object == this))
         {
-            JSON_THROW(invalid_iterator::create(211, "passed iterators may not belong to container"));
+            JSON_THROW(invalid_iterator::create(211, "passed iterators may not belong to container", diagnostics_t(*this)));
         }
 
         // insert to array and return iterator
+#if JSON_DIAGNOSTICS
+        iterator result = insert_iterator(pos, first.m_it.array_iterator, last.m_it.array_iterator);
+        for (typename iterator::difference_type i = 0; i < std::distance(first, last); ++i)
+        {
+            (result + i)->m_parent = this;
+        }
+        return result;
+#else
         return insert_iterator(pos, first.m_it.array_iterator, last.m_it.array_iterator);
+#endif
```

Comment:
As above

        iterator result = insert_iterator(pos, first.m_it.array_iterator, last.m_it.array_iterator);
    #if JSON_DIAGNOSTICS
        for (typename iterator::difference_type i = 0; i < std::distance(first, last); ++i)
        {
            set_parent(result + i);
        }
    #endif
        return result;

Reply:
Fixed with f8037660d06a0fbb6b1cde944d436032d04564ae.